File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/java/io/github/lishangbu/avalon/dataset/entity/PokemonType.java
package io.github.lishangbu.avalon.dataset.entity;

import jakarta.persistence.*;
import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/// 宝可梦属性(PokemonType)实体类
///
/// 表示宝可梦与属性的关联信息，包括排序信息
///
/// @author lishangbu
/// @since 2025/08/20
@Data
@Entity
@Table(comment = "宝可梦属性")
public class PokemonType implements Serializable {
  @Serial private static final long serialVersionUID = 1L;

  /// 复合主键
  @EmbeddedId private PokemonTypeId id;

  /// 属性内部slot
  @Column(comment = "属性内部slot")
  private Integer slot;

  /// 宝可梦属性复合主键
  ///
  /// @author lishangbu
  /// @since 2026/2/16
  @Data
  @Embeddable
  public static class PokemonTypeId implements Serializable {
    @Serial private static final long serialVersionUID = 1L;

    /// 宝可梦引用
    @ManyToOne
    @JoinColumn(name = "pokemon_id", nullable = false)
    private Pokemon pokemon;

    /// 属性引用
    @ManyToOne
    @JoinColumn(name = "type_id", nullable = false)
    private Type type;
  }
}

File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/kotlin/io/github/lishangbu/avalon/dataset/entity/PokemonType.kt
package io.github.lishangbu.avalon.dataset.entity

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.PropOverride

@Entity
interface PokemonType {
    /** ID */
    @Id
    @PropOverride(prop = "pokemonId", columnName = "pokemon_id")
    @PropOverride(prop = "typeId", columnName = "type_id")
    val id: PokemonTypeId

    /** 槽位 */
    val slot: Int?
}
